src/DurableTask.Netherite/Scaling/AzureTableLoadPublisher.cs

```diff
@@ -13,104 +16,84 @@ namespace DurableTask.Netherite.Scaling
 
     class AzureTableLoadPublisher : ILoadPublisherService
     {
-        readonly CloudTable table;
+        readonly TableClient table;
         readonly string taskHubName;
 
         public AzureTableLoadPublisher(string connectionString, string tableName, string taskHubName)
         {
-            var account = CloudStorageAccount.Parse(connectionString);
-            this.table = account.CreateCloudTableClient().GetTableReference(tableName);
+            this.table = new TableClient(connectionString, tableName); 
             this.taskHubName = taskHubName;
         }
 
         public TimeSpan PublishInterval => TimeSpan.FromSeconds(10);
 
         public async Task DeleteIfExistsAsync(CancellationToken cancellationToken)
         {
-            if (! await this.table.ExistsAsync().ConfigureAwait(false))
+            try
             {
-                return;
+                var tableBatch = new List<TableTransactionAction>();
+                await foreach (var e in this.table.QueryAsync<PartitionInfoEntity>($"PartitionKey eq '{this.taskHubName}'"))
+                {
+                    tableBatch.Add(new TableTransactionAction(TableTransactionActionType.Delete, e));
+                }
+                await this.table.SubmitTransactionAsync(tableBatch, cancellationToken);
             }
-
-            var query = new TableQuery<PartitionInfoEntity>().Where(TableQuery.GenerateFilterCondition("RowKey", QueryComparisons.Equal, this.taskHubName));
-            TableContinuationToken continuationToken = null;
-            do
+            catch(Exception) // table may not exist
             {
-                var batch = await this.table.ExecuteQuerySegmentedAsync<PartitionInfoEntity>(query, continuationToken, null, null, cancellationToken).ConfigureAwait(false);
 
-                if (batch.Count() > 0)
-                {
-                    // delete all entities in this batch. Max partition number is 32 so it always fits.
-                    TableBatchOperation tableBatch = new TableBatchOperation();
-
-                    foreach (var e in batch)
-                    {
-                        tableBatch.Add(TableOperation.Delete(e));
-                    }
-
-                    await this.table.ExecuteBatchAsync(tableBatch).ConfigureAwait(false);
-                }
             }
-            while (continuationToken != null);
         }
 
-        public async Task CreateIfNotExistsAsync(CancellationToken cancellationToken)
+        public Task CreateIfNotExistsAsync(CancellationToken cancellationToken)
         {
-            if (await this.table.ExistsAsync().ConfigureAwait(false))
-            {
-                return;
-            }
-
-            await this.table.CreateAsync().ConfigureAwait(false);
+            return this.table.CreateIfNotExistsAsync();
```

Comment:
pass in cancellation token:
```suggestion
            return this.table.CreateIfNotExistsAsync(cancellationToken);
```

Reply:
These days I try to remove the ConfigureAwait(false) noise. All my code is running in tasks on the thread pool, there is no implicit synchronization context to worry about.

Reply:
Possibly so. I will take another look. Usually I don't bother because most of my code is only ever called from thread pool tasks, but this may be called from some place that has an implicit synchronization context.